cpp/src/arrow/acero/swiss_join_internal.h

```diff
@@ -161,7 +161,7 @@ class RowArrayMerge {
   //
   static Status PrepareForMerge(RowArray* target, const std::vector<RowArray*>& sources,
                                 std::vector<int64_t>* first_target_row_id,
-                                MemoryPool* pool);
+                                MemoryPool* pool, bool check_key_size);
```

Comment:
Nits:
1. let's keep the `MemoryPool*` argument at the end
2. perhaps change the argument name to not expose an implementation detail?

So for example:
```c++
  static Status PrepareForMerge(RowArray* target, const std::vector<RowArray*>& sources,
                                bool is_key_data, std::vector<int64_t>* first_target_row_id,
                                MemoryPool* pool);
```

Reply:
Thanks for your review! I think I adjusted my PR accordingly.